modules/ui/commit_warnings.js

```diff
@@ -1,103 +1,113 @@
             .getIssuesBySeverity({ what: 'edited', where: 'all', includeDisabledRules: true });
 
-        for (var severity in issuesBySeverity) {
-            var issues = issuesBySeverity[severity];
+        for (let severity in _issuesBySeverity) {
+            let issues = _issuesBySeverity[severity];
 
-            if (severity !== 'error') {      // exclude 'fixme' and similar - #8603
+            if (severity !== 'error') { // exclude 'fixme' and similar - #8603
                 issues = issues.filter(function(issue) { return issue.type !== 'help_request'; });
             }
 
-            var section = severity + '-section';
-            var issueItem = severity + '-item';
-
-            var container = selection.selectAll('.' + section)
-                .data(issues.length ? [0] : []);
-
-            container.exit()
-                .remove();
-
-            var containerEnter = container.enter()
-                .append('div')
-                .attr('class', 'modal-section ' + section + ' fillL2');
-
-            containerEnter
-                .append('h3')
-                .call(severity === 'warning' ? t.append('commit.warnings') : t.append('commit.errors'));
-
-            containerEnter
-                .append('ul')
-                .attr('class', 'changeset-list');
-
-            container = containerEnter
-                .merge(container);
-
+            if (!issues.length) continue;
 
-            var items = container.select('ul').selectAll('li')
-                .data(issues, function(d) { return d.key; });
-
-            items.exit()
-                .remove();
-
-            var itemsEnter = items.enter()
-                .append('li')
-                .attr('class', issueItem);
-
-            var buttons = itemsEnter
-                .append('button')
-                .on('mouseover', function(d3_event, d) {
-                    if (d.entityIds) {
-                        context.surface().selectAll(
-                            utilEntityOrMemberSelector(
-                                d.entityIds,
-                                context.graph()
-                            )
-                        ).classed('hover', true);
-                    }
+            // Create a collapsible section for each severity level
+            var section = uiSection('issues-' + severity, context)
+                .label(() => {
+                    var count = issues.length;
+                    return t.append(
+                        'inspector.title_count',
+                        { title: t('issues.' + severity + 's.list_title'), count: count }
+                    );
                 })
-                .on('mouseout', function() {
-                    context.surface().selectAll('.hover')
-                        .classed('hover', false);
+                .disclosureContent(function(selection) {
+                    return renderIssuesList(selection, severity, issues);
                 })
```

Comment:
this could be an arrow function to be more concise and consistent with the `.label()` property a few lines above 

```suggestion
                .disclosureContent(selection => renderIssuesList(selection, ~~severity,~~ issues))
```

same also for the `shouldDisplay` callback and several ones called in `renderIssuesList`.

Reply:
i've tried all of the above things but there's problem occuring 
![image](https://github.com/user-attachments/assets/d133d7e1-dfaf-4238-b3b3-af0ceaebdcc3)
when i hit save commitWarnings() function is called twice and when click anywhere in the screen again it called and new sections of warnings (duplicate sections) are being added at the bottom 
@tyrasd can you help me with this

Reply:
applied all the requested changes
![image](https://github.com/user-attachments/assets/4cf851eb-036f-4532-923e-0a2016292772)
@tyrasd need to review

Reply:
@tyrasd if you are not busy could you please review it